lib/bn.js

```diff
@@ -56,7 +56,7 @@
 
   BN.isBN = function isBN (num) {
     return num !== null && typeof num === 'object' &&
-      num.constructor.name === 'BN' && Array.isArray(num.words);
+      num.constructor.name === BN.name && Array.isArray(num.words);
```

Comment:
You know what I learned recently? This thing does not work in IE: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name .

Looks like we may need to fallback to using just `Array.isArray()`.